docs/Training-on-Amazon-SageMaker-RL.md

```diff
@@ -0,0 +1,283 @@
+
+```
+import sagemaker
+import boto3
+
+# set up the linkage and authentication to the S3 bucket
+sage_session = sagemaker.session.Session()
+s3_bucket = sage_session.default_bucket()
+s3_output_path = 's3://{}/'.format(s3_bucket)
+print("S3 bucket path: {}".format(s3_output_path))
+
+# create a descriptive job name
+job_name_prefix = 'rl-unity-ray'
+
+# configure where training happens – local or SageMaker instance
+local_mode = False
+
+if local_mode:
+    instance_type = 'local'
+else:
+    # If on SageMaker, pick the instance type
+    instance_type = "ml.c5.2xlarge"
+
+# create an IAM role
+try:
+    role = sagemaker.get_execution_role()
+except:
+    role = get_execution_role()
+
+print("Using IAM role arn: {}".format(role))
+```
+
+### Building a Docker container
+
+SageMaker uses Docker containers to run scripts, train algorithms, and deploy models. A [Docker container](https://www.docker.com/resources/what-container) is a standalone package of software that manages all the code and dependencies, and it includes everything needed to run an application. We start by building on top of a pre-built SageMaker Docker image that contains dependencies for Ray, then install the required core packages:
+
+* `gym-unity` – Unity provides a wrapper to wrap Unity environment into a [gym](https://github.com/openai/gym) interface, an open-source library that gives you access to a set of classic RL environments
+* `mlagents-envs` – Package that provides a Python API to allow direct interaction with the Unity game engine
+
+For all pre-built SageMaker RL Docker images, see the [GitHub repo](https://github.com/aws/sagemaker-rl-container).
+
+```
+ARG CPU_OR_GPU
+ARG AWS_REGION
+# Specify a SageMaker RL Docker image
+FROM 462105765813.dkr.ecr.${AWS_REGION}.amazonaws.com/sagemaker-rl-ray-container:ray-0.8.2-tf-${CPU_OR_GPU}-py36
+
+WORKDIR /opt/ml
+
+# Unity dependencies
+
+RUN pip install --upgrade \
+    pip \
+    gym-unity \
+    mlagents-envs
+
+RUN pip install sagemaker-containers --upgrade
+
+ENV PYTHONUNBUFFERED 1
+
+############################################
+# Test Installation
+############################################
+# Test to verify if all required dependencies installed successfully or not.
+RUN python -c "import gym;import sagemaker_containers.cli.train;import ray; from sagemaker_containers.cli.train import main; from mlagents_envs.environment import UnityEnvironment; from mlagents_envs.registry import default_registry; from gym_unity.envs import UnityToGymWrapper"
+
+# Make things a bit easier to debug
+WORKDIR /opt/ml/code
+```
+
+### Preparing your Unity environment
+
+1. In the Unity Editor, load a project containing an ML-Agents environment (you can use one of the example environments if you have not created your own but please make sure your environment only contains one single agent).
```

Comment:
I would make a note that this example uses the Basic environment and link it (https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Learning-Environment-Examples.md#basic)